core/src/main/java/cucumber/runner/TestCaseSyncEventBus.java

```diff
@@ -0,0 +1,69 @@
+package cucumber.runner;
+
+import cucumber.api.event.Event;
+import cucumber.api.event.EventHandler;
+import cucumber.api.event.TestCaseEvent;
+import cucumber.api.event.TestCaseFinished;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestCaseSyncEventBus implements EventBus {
+
+    // Static sync object so all instances can be locked together 
+    private static final Object SYNC = new Object();
+    private static final Class<? extends Event> FLUSH_EVENT = TestCaseFinished.class;
+
+    private final ThreadLocal<List<Event>> queue = new ThreadLocal<List<Event>>() {
+        @Override
+        protected List<Event> initialValue() {
```

Comment:
Java  1.8

Reply:
Again it’s in javadocs earlier than 1.8, plus IntelliJ isn’t moaning and it has done so when i’ve to use methods / syntax from later than 1.7

I’ll just change it back anyway

Reply:
You can change that setting.

It sucks. I know.

Reply:
As you said earlier you were getting confused with `ThreadLocal.withInitial`.  No changes here